### ISSUE TITLE
DatabaseErrorPageMiddleware fails when hosted in IIS sub-application

### ISSUE THREAD

**Issue by ArturDorochowicz** [dotnet/aspnetcore#22517]
### Describe the bug

When `DatabaseErrorPageMiddleware` generates an apply migrations web page, it outputs `DatabaseErrorPageOptions.MigrationsEndPointPath` as is for the client side javascript to call. This value (by default `/ApplyDatabaseMigrations`) is not adjusted to take into consideration the `PathBase` of the application.

If the application is hosted in IIS sub-application at `/sub/`, the correct url to call would be `/sub/ApplyDatabaseMigrations`, but what is called instead is `/ApplyDatabaseMigrations` (which does not exist).

Problematic code is here:
https://github.com/dotnet/aspnetcore/blob/v3.1.4/src/Middleware/Diagnostics.EntityFrameworkCore/src/Views/DatabaseErrorPage.cshtml#L98

### To Reproduce
1. Prepare the project
```
dotnet new web -n DatabaseErrorPage
cd DatabaseErrorPage
dotnet add package Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore
dotnet add package Microsoft.EntityFrameworkCore.Sqlite
dotnet add package Microsoft.EntityFrameworkCore.Design
dotnet new tool-manifest
dotnet tool install dotnet-ef
```
2. Modify `launchSettings.json` for sub-application
```json
{
  "iisSettings": {
    "windowsAuthentication": false, 
    "anonymousAuthentication": true, 
    "iisExpress": {
      "applicationUrl": "http://localhost:36046/sub",
      "sslPort": 44381
    }
  },
  "profiles": {
    "IIS Express": {
      "commandName": "IISExpress",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "DatabaseErrorPage": {
      "commandName": "Project",
      "launchBrowser": true,
      "applicationUrl": "https://localhost:5001;http://localhost:5000",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
  }
}
```
3. Modify `Startup.cs`
```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Mime;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Http;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace DatabaseErrorPage
{
    public class Startup
    {
        // This method gets called by the runtime. Use this method to add services to the container.
        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddDbContext<AppDb>(
                _ => _.UseSqlite("Data Source=:memory:;"));
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
                app.UseDatabaseErrorPage();
            }

            app.UseRouting();

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapGet("/", async context =>
                {
                    // trigger database exception
                    await context.RequestServices.GetRequiredService<AppDb>().Blogs.CountAsync();

                    await context.Response.WriteAsync("Hello World!");
                });
            });
        }
    }

    public class AppDb : DbContext
    {
        public AppDb(DbContextOptions<AppDb> options)
            : base(options)
        {
        }

        public DbSet<Blog> Blogs { get; set; }
    }

    public class Blog
    {
        public Guid Id { get; set; }
    }
}
```
4. Add a migration
```
dotnet ef migrations add Initial
```
5. Run the project in VS

### Further technical details
```
$> dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.300
 Commit:    b2475c1295

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.300/

Host (useful for support):
  Version: 3.1.4
  Commit:  0090613580

.NET Core SDKs installed:
  3.1.300 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by shirhatti** [comment]
@ajcvickers Can you have a look?

**Comment by ajcvickers** [comment]
@shirhatti I don't know much about middleware. This sounds like a bug from the description by the customer, but can someone who actually knows how middleware works confirm this?

**Comment by BrennanConroy** [comment]
The link that is generated needs to include the `Request.PathBase`

**Comment by JunTaoLuo** [comment]
We have decided to remove this feature so this bug will also no longer exist.

**Comment by ArturDorochowicz** [comment]
What about 2.1 and 3.1 which are still supported?

pt., 3 lip 2020, 00:53 użytkownik John Luo <notifications@github.com>
napisał:

**Comment by JunTaoLuo** [comment]
I'm not sure if this meets the servicing bar for 2.1 and 3.1 thoughts @ajcvickers ?

**Comment by ajcvickers** [comment]
@JunTaoLuo Pretty sure this doesn't meet the bar.

**Comment by JunTaoLuo** [comment]
We decided to retain the functionality so I'll fix the issue by including pathbase in 5.0.0-rc1.

**Comment by JunTaoLuo** [comment]
The fix has been merged for 5.0.0-rc1.
